Honeycomb Bravo.lua

```diff
@@ -116,396 +133,617 @@ function array_has_true(array)
-						-- Gear moving
-						gear_leds[i][1] = false
-						gear_leds[i][2] = true
-					end
+					-- Gear moving
+					gear_leds[i][1] = false
+					gear_leds[i][2] = true
 				end
 			end
+		end
 
-			set_led(LED.LDG_N_GREEN, gear_leds[1][1])
-			set_led(LED.LDG_N_RED, gear_leds[1][2])
-			set_led(LED.LDG_L_GREEN, gear_leds[2][1])
-			set_led(LED.LDG_L_RED, gear_leds[2][2])
-			set_led(LED.LDG_R_GREEN, gear_leds[3][1])
-			set_led(LED.LDG_R_RED, gear_leds[3][2])
-
-			-- MASTER WARNING
-			set_led(LED.ANC_MSTR_WARNG, int_to_bool(master_warn[0]))
-
-			-- ENGINE FIRE
+		set_led(LED.LDG_N_GREEN, gear_leds[1][1])
+		set_led(LED.LDG_N_RED, gear_leds[1][2])
+		set_led(LED.LDG_L_GREEN, gear_leds[2][1])
+		set_led(LED.LDG_L_RED, gear_leds[2][2])
+		set_led(LED.LDG_R_GREEN, gear_leds[3][1])
+		set_led(LED.LDG_R_RED, gear_leds[3][2])
+
+		-- MASTER WARNING
+		set_led(LED.ANC_MSTR_WARNG, int_to_bool(master_warn[0]))
+
+		-- ENGINE/APU FIRE
+		if PROFILE == 'Toliss/32x' then
+			my_fire = false
+			if apu_fire[20] > 0 then
+				my_fire = true
+			end
+			for i = 11, 17 do
+				if i == 11 or i == 13 or i == 15 or i == 17 then
+					if eng_fire[i] ~= nil and eng_fire[i] > 0 then
+						my_fire = true
+						break
+					end
+				end
+			end
+			set_led(LED.ANC_ENG_FIRE, my_fire)
+		else
 			set_led(LED.ANC_ENG_FIRE, array_has_true(fire))
+		end
 
-			-- LOW OIL PRESSURE
+		-- LOW OIL PRESSURE
+		if PROFILE == "Toliss/32x" then
+			low_oil_light = false
+			for i = 0, NUM_ENGINES - 1 do
+				if oil_low_p[i] ~= nil and oil_low_p[i] < 0.075 then
+					low_oil_light = true
+					break
+				end
+			end
+			set_led(LED.ANC_OIL, low_oil_light)
+		else
 			set_led(LED.ANC_OIL, array_has_true(oil_low_p))
+		end
 
-			-- LOW FUEL PRESSURE
+		-- LOW FUEL PRESSURE
+		if PROFILE == "Toliss/32x" then
+			low_fuel_light = false
+			for i = 0, NUM_ENGINES - 1 do
+				if fuel_low_p[i] ~= nil and fuel_low_p[i] < 0.075 then
+					low_fuel_light = true
+					break
+				end
+			end
+			set_led(LED.ANC_FUEL, low_fuel_light)
+		else
 			set_led(LED.ANC_FUEL, array_has_true(fuel_low_p))
+		end
 
-			-- ANTI ICE
-			if not anti_ice_flip then
-				set_led(LED.ANC_ANTI_ICE, int_to_bool(anti_ice[0]))
-			else
-				set_led(LED.ANC_ANTI_ICE, not int_to_bool(anti_ice[0]))
-			end
+		-- ANTI ICE
+		if not anti_ice_flip then
+			set_led(LED.ANC_ANTI_ICE, array_has_positives(anti_ice, 10))
+		else
+			set_led(LED.ANC_ANTI_ICE, not array_has_positives(anti_ice, 10))
+		end
 
-			-- STARTER ENGAGED
-			set_led(LED.ANC_STARTER, array_has_true(starter))
+		-- STARTER ENGAGED
+		set_led(LED.ANC_STARTER, array_has_true(starter))
 
-			-- APU
-			set_led(LED.ANC_APU, int_to_bool(apu[0]))
+		-- APU
+		set_led(LED.ANC_APU, int_to_bool(apu[0]))
 
-			-- MASTER CAUTION
-			set_led(LED.ANC_MSTR_CTN, int_to_bool(master_caution[0]))
+		-- MASTER CAUTION
+		set_led(LED.ANC_MSTR_CTN, int_to_bool(master_caution[0]))
 
-			-- VACUUM
+		-- VACUUM
+		if PROFILE == "Toliss/32x" then
+			if vacuum[0] < 1 then
```

Comment:
Is the vacuum dataref flipped for the Toliss? If that's the case then I think we should use something like the `anti_ice_flip` rather than this logic; so there isn't so much custom code for future aircraft.